process/smartContract/hooks/blockChainHook.go

```diff
@@ -404,6 +422,26 @@ func (bh *BlockChainHookImpl) NewAddress(creatorAddress []byte, creatorNonce uin
 func (bh *BlockChainHookImpl) ProcessBuiltInFunction(input *vmcommon.ContractCallInput) (*vmcommon.VMOutput, error) {
 	defer stopMeasure(startMeasure("ProcessBuiltInFunction"))
 
+	if bh.enableEpochsHandler.IsMaxBlockchainHookCountersFlagEnabled() {
+		gasSchedule := bh.gasSchedule.LatestGasSchedule()
+
+		maxBuiltInCallsPerTx := gasSchedule["MaxPerTransaction"]["MaxBuiltInCallsPerTx"]
+		bh.crtNumberOfBuiltInFunctionCalls++
+		if bh.crtNumberOfBuiltInFunctionCalls > maxBuiltInCallsPerTx {
+			return nil, process.OutOfAPICalls
+		}
+
+		maxNumberOfTransfersPerTx := gasSchedule["MaxPerTransaction"]["MaxNumberOfTransfersPerTx"]
+		parsedTransfer, err := bh.esdtTransferParser.ParseESDTTransfers(input.CallerAddr, input.RecipientAddr, input.Function, input.Arguments)
+		if err != nil {
+			return nil, err
+		}
+		bh.crtNumberOfTransfers += uint64(len(parsedTransfer.ESDTTransfers))
+		if bh.crtNumberOfTransfers > maxNumberOfTransfersPerTx {
+			return nil, process.OutOfAPICalls
```

Comment:
this whole code can be move into a new function. from 425-444

Reply:
Move the whole part of nr calls and nr transfer checks to line 458 - exactly before calling the ProcessBuiltInFunction